src/pint_pal/timingconfiguration.py

```diff
@@ -714,6 +720,11 @@ def apply_ignore(self,toas,specify_keys=None,warn=False,model=None):
             for o in self.get_orphaned_rec():
                 orphinds = np.where(fs==o)[0]
                 apply_cut_flag(toas,orphinds,'orphaned',warn=warn)
+        if 'bad-group' in valid_valued:
+            for flag, value in self.get_bad_group():
+                vals = np.array([f[flag] for f in toas.orig_table['flags']])
```

Comment:
I just found for the IPTA DR3 data, this can error for cases where some ToAs have flags other ToAs don't (e.g. if InPTA uses 'group' and NANOGrav doesn't, specifying 'group' as a bad-group throws an error when hitting the NANOGrav ToA). A fix for this is to add something like `[f[flag] for f in toas.orig_table['flags'] if flag in list(f.keys())]` to check if the flag exists

Reply:
Here I was just copying how the other excision keys (such as "orphaned-rec" or "poor-febe") work, which means that they probably have the same issue. For now I'll just fix this for "bad-group". If we need to change the others, that can go into a new MR.